File: RepeatedDNASequences.cpp
class Solution {
public:
    vector<string> findRepeatedDnaSequences(string s) {
        unordered_map <string,int> table;
        vector <string> result;

        if(s.length()<=10)
            return result;
        
        for(int i=0; i<s.length()-10+1; i++)
            if(++table[s.substr(i,10)]==2)
                result.push_back(s.substr(i,10));
    
        return result;
    }
};

File: RepeatedDNASequences.py
class Solution:
    def findRepeatedDnaSequences(self, s: str) -> List[str]:
        table={}
        for i in range(len(s)-10+1): # push all substring
            value=s[i:i+10]
            if(value in table):
                table[value]+=1
            else:
                table[value]=1
                
        return [p for p in table.keys() if(table[p]>1)]
